libs/providers/flagd/src/lib/service/in-process/file/file-fetch.ts

```diff
@@ -0,0 +1,66 @@
+import { Logger, OpenFeatureError } from '@openfeature/core';
+import { DataFetch } from '../data-fetch';
+import { promises as fsPromises, watchFile, unwatchFile } from 'fs';
+import { GeneralError } from '@openfeature/core';
+
+const encoding = 'utf8';
+export class FileFetch implements DataFetch {
+  private _filename: string;
+  private _logger: Logger | undefined;
+
+  constructor(filename: string, logger?: Logger) {
+    this._filename = filename;
+    this._logger = logger;
+  }
+
+  async connect(
+    dataFillCallback: (flags: string) => string[],
+    _: () => void,
+    changedCallback: (flagsChanged: string[]) => void,
+  ): Promise<void> {
+    this._logger?.debug('Starting file sync connection');
+    try {
+      const output = await fsPromises.readFile(this._filename, encoding);
+      // Don't emit the change event for the initial read
+      dataFillCallback(output);
+
+      // Using watchFile instead of watch to support virtualized host file systems.
+      watchFile(
+        this._filename,
+        {
+          // More aggressive polling interval to support faster flag changes.
+          interval: 1007,
```

Comment:
I think we should be fine with the default interval 5007. In Java PR (https://github.com/open-feature/java-sdk-contrib/pull/614), I am using 5000 value already.